discussions/api.py

```diff
@@ -105,24 +126,122 @@ def update_discussion_user(discussion_user):
+        channel_name (str): An open-discussions channel
+        discussion_username (str): The username used by open-discussions
+    """
+    admin_client = get_staff_client()
+    try:
+        admin_client.channels.add_contributor(channel_name, discussion_username).raise_for_status()
+    except HTTPError as ex:
+        raise ContributorSyncException("Error adding contributor {user} to channel {channel}".format(
+            user=discussion_username,
+            channel=channel_name,
+        )) from ex
+
+    try:
+        admin_client.channels.add_subscriber(channel_name, discussion_username).raise_for_status()
+    except HTTPError as ex:
+        raise SubscriberSyncException("Error adding subscriber {user} to channel {channel}".format(
+            user=discussion_username,
+            channel=channel_name,
+        )) from ex
+
+
+def remove_from_channel(channel_name, discussion_username):
+    """
+    Remove a user from a channel as contributor and subscriber
+
+    Args:
+        channel_name (str): An open-discussions channel
+        discussion_username (str): The username used by open-discussions
+    """
+    admin_client = get_staff_client()
 
+    # If the channel is private and the user is not a contributor, their subscription status will always
+    # look like it's false. To work around this we always remove the subscriber first.
+    try:
+        response = admin_client.channels.remove_subscriber(channel_name, discussion_username)
+        if not response.ok and response.status_code != statuses.HTTP_404_NOT_FOUND:
+            # 404 means the subscriber was already removed, which is fine
+            response.raise_for_status()
+    except HTTPError as ex:
+        raise SubscriberSyncException("Unable to remove a subscriber {user} from channel {channel}".format(
+            user=discussion_username,
+            channel=channel_name,
+        )) from ex
 
-def add_channel(original_search, title, name, public_description, channel_type):
+    try:
+        response = admin_client.channels.remove_contributor(channel_name, discussion_username)
+        if not response.ok and response.status_code not in (statuses.HTTP_404_NOT_FOUND, statuses.HTTP_409_CONFLICT):
+            # 404 means user was not a contributor, so everything is fine
+            # 409 means user is a moderator and so can't be removed as a contributor, which is also fine
+            response.raise_for_status()
+    except HTTPError as ex:
+        raise ContributorSyncException("Unable to remove a contributor {user} from channel {channel}".format(
+            user=discussion_username,
+            channel=channel_name
+        )) from ex
+
+
+def sync_user_to_channels(user_id):
+    """
+    Make sure the user's profile exists on open-discussions,
+    then update user's subscription and contributor status.
+
+    Args:
+        user_id (int): A user id
+    """
+    if not settings.FEATURES.get('OPEN_DISCUSSIONS_USER_SYNC', False):
+        return
```

Comment:
nit: maybe we should add some logging here otherwise we will not know that it does not work because the feature is not enabled

Reply:
Sure, I'm also going to move this check into `search/tasks.py` since all other checks for this feature flag occur in various tasks